CMakeLists.txt

```diff
@@ -68,6 +68,24 @@ if(NOT CMAKE_CXX_STANDARD)
 endif()
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+include(CheckCXXCompilerFlag)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  check_cxx_compiler_flag(-fcoroutines COMPILER_HAS_F_COROUTINES)
+  if (COMPILER_HAS_F_COROUTINES)
+    message(
+      STATUS
+      "GCC has support for C++ coroutines, setting flag for FBThrift build."
+    )
+    add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fcoroutines>)
+  else()
```

Comment:
Note that gcc >= 10 supports coroutines: https://godbolt.org/z/Tz8TTe1MW.

And that folly, a key dependency of fbrthrift, requires gcc => gcc >= 10: https://github.com/facebook/folly/blob/v2024.06.24.00/folly/Portability.h#L37-L39.

So we should not actually need this `else`. We mostly don't even need the conditional - at best, it protects against some hypothetical future version of gcc removing the feature switch `-fcoroutines`.

Reply:
Thanks, this is good to know!

I'm now having some second thoughts about enabling the flag unconditionally—perhaps it'd be less risky to do nothing on the fbthrift side and just have C++20 consumers set `-fno-coroutines` in their own projects instead.

Reply:
Note that coroutines is just a part of the language and library now, as of C++20. It is not this extra, separate thing that has to be opted into - except in C++17, which we still partially support for a limited time, and except in GCC = 10, which doesn't automatically enable coroutines. From a forward-looking perspective, coroutines will be just another tool.

I would be in favor of setting `-fcoroutines` when building with GCC >= 11 and C++ = 17 or when building with GCC = 10. That way, all builds uniformly have coroutines enabled. Once GCC >= 11 is the baseline and C++20 is the baseline, that's the way it will be anyway.

Reply:
Makes sense, thanks for clarifying!